bftengine/src/bftengine/ReadOnlyReplica.cpp

```diff
@@ -244,11 +247,14 @@ void ReadOnlyReplica::executeReadOnlyRequest(concordUtils::SpanWrapper &parent_s
       send(&reply, clientId);
     } else {
       LOG_WARN(GL, "Received zero size response. " << KVLOG(clientId));
+      executionResult = bftEngine::OperationResult::EMPTY_EXEC_DATA;
     }
 
   } else {
     LOG_ERROR(GL, "Received error while executing RO request. " << KVLOG(clientId, status));
   }
+  ClientReplyMsg replyMsg(0, request.requestSeqNum(), config_.getreplicaId(), executionResult);
+  send(&replyMsg, clientId);
```

Comment:
I still see two replies get sent here: in lines 244 and 257. The last one gets sent unconditionally, not only when status != 0.